### PR TITLE
Add support for more operations

### PR DESCRIPTION
**Pull request by jurajsic**
This PR adds support for more string operations, specifically:
- regex complementation
- regex intersection
- regex ".+" which is in z3 represented specifically
- empty language
- language containing only epsilon
- counting loops "R{n,m}"

It also add function `throw_error` which we should use whenever we want to throw error in noodler. Based on debugging, it either throws std::runtime_error, which is not caught by Z3, so we can see what the problem is while debugging, otherwise it throws some internal Z3 exception, which Z3 can catch and return unknown.

I will run it on the benchmarks if it works correctly and then we can merge.

### CODE REVIEW

**Review by vhavlena — APPROVED** [review]

**Comment by jurajsic** [comment]
I run it on all benchmarks, everything was solved correctly (same as cvc5), but for no apparent reason, there is degradation of speed for one formula from slent benchmark (`slent_stranger_str_rep_21602_sink.smt2`) and the speed for leetcode benchmark jumps a bit:
image

File: src/smt/theory_str_noodler/util.cc [hunk 1]
```diff
@@ -466,10 +535,10 @@ namespace smt::noodler::util {
                 //  Solution: Unify initial and add the unified initial as a final.
                 nfa.final.add(initial);
```
Comment (Adda0) [inline src/smt/theory_str_noodler/util.cc:536]: We can now remove these FIXME and Solution comments as the initial states are already unified, and the mentioned situation cannot occur.

**Review by Adda0 — APPROVED** [review]
The changes seem good to me.

File: src/smt/theory_str_noodler/util.cc (hunk 1 above)
Comment (jurajsic) [reply src/smt/theory_str_noodler/util.cc]: removed

**Pull request merged**